### ISSUE TITLE
Prevent a node from becoming a "folder" when a node is dropped on it

### ISSUE THREAD

**Issue by jago86** [amsik/liquor-tree#72]
Hi, I'm working with dnd. Normally when I dnd over an node that have no children, it becomes in a node with children (I call it *folder* in my specific case). 

![become_folder](https://user-images.githubusercontent.com/1690276/48106341-09b15600-e209-11e8-81e5-91fc16ce1b0e.gif)

So, I know that I can use the `state: { dropable: false },` option. The problem is when I apply this to all my nodes with no children and the parent node is expanded, there is no way to drop any node inside, is like thers's no droppable area available.

![cant_drop](https://user-images.githubusercontent.com/1690276/48106142-3e70dd80-e208-11e8-9a27-53db231bc134.gif)

Is there any way to handle this? Thanks in advance.

**Comment by amsik** [comment]
Hi. Yes, it is a bug. I will fix it as soon as possible. Thanks for the example

**Comment by jago86** [comment]
Thanks @amsik !!! Also, I'm think there is a related problem. When the parent node runs out of children, it becomes a *file* (node with no children). I think it should stay as *folder* (node with children) even if has no children.

![becomes_file](https://user-images.githubusercontent.com/1690276/48143367-99441c80-e27c-11e8-8dbe-eabfc9a0bc7d.gif)

**Comment by amsik** [comment]
I don't agree with you. For your case I would suggest to use `data` property to check the type.

**Comment by jago86** [comment]
Ok. You're right, I'll use the data property. Then this issue is only about the first message in this thread.

**Comment by amsik** [comment]
Fixed. I will add it to the npm a little bit later

**Comment by jago86** [comment]
Thanks man!!! If you don't mind, let us know when you update in npm. Thanks again.

**Comment by amsik** [comment]
Updated
